app/src/main/java/com/github/onlynotesswent/model/ocr/CustomTextRecognizer.kt

```diff
@@ -0,0 +1,104 @@
+package com.github.onlynotesswent.model.ocr
+
+import android.content.Intent
+import android.net.Uri
+import android.util.Log
+import android.widget.Toast
+import androidx.activity.ComponentActivity
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import com.google.mlkit.vision.common.InputImage
+import com.google.mlkit.vision.text.TextRecognition
+import com.google.mlkit.vision.text.TextRecognizer
+import com.google.mlkit.vision.text.latin.TextRecognizerOptions
+import java.io.IOException
+
+/* Using Google ML Kit’s Text Recognition v2 API to extract text from an image:
+  https://developers.google.com/ml-kit/vision/text-recognition/v2
+ Inspired by: https://github.com/mundodigitalpro/MLVisionKotlin/tree/master
+*/
+
+/**
+ * CustomTextRecognizer class that detects latin-based text from an input image
+ *
+ * @param activity the ComponentActivity that will use the CustomTextRecognizer
+ * @param textRecognizer the TextRecognizer object, initialized with the default options
+ */
+class CustomTextRecognizer(
+    private val activity: ComponentActivity,
+    private val textRecognizer: TextRecognizer =
+        TextRecognition.getClient(TextRecognizerOptions.DEFAULT_OPTIONS)
+) {
+
+  private lateinit var textRecognitionLauncher: ActivityResultLauncher<String>
+
+  /**
+   * Initializes the activity result launcher to handle the image result. To be called in the
+   * `onCreate` method of the activity.
+   */
+  fun init() {
+    textRecognitionLauncher =
+        activity.registerForActivityResult(ActivityResultContracts.GetContent()) { uri ->
+          uri?.let { extractTextFromImage(it) }
+              ?: Toast.makeText(activity, "Failed to load image", Toast.LENGTH_LONG).show()
+        }
+  }
+
+  /** Launches an intent to pick an image from the gallery. */
+  fun scanImage() {
+    textRecognitionLauncher.launch("image/*")
```

Comment:
Catch ActivityNotFoundException, and maybe add a check to make sure textRecognitionLauncher was initialized (init was correctly called before a scan is attempted, this check shouldn't happen if code is correct so only a log of error is enough)

Reply:
Good catch, thank you